### PR TITLE
use getSOCDMode in DDI None to benefit from 28d47d9f

### PR DESCRIPTION
**Pull request by bsstephan**
Just a tiny code cleanup/improvement to my previous PR to account for the no SOCD mode change that also went in.

### CODE REVIEW

File: src/addons/dualdirectional.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ void DualDirectionalInput::process()
             // this also avoids accidentally masking gamepad inputs with the lack of dual inputs
             if (gamepad->options.dpadMode == options.dualDirDpadMode) {
```
Comment (deeebug) [inline src/addons/dualdirectional.cpp:158]: I've done some tests and this seems to be the logic that gets both regular and DDI working together when both input methods are same.

```c
if (gamepad->options.dpadMode == options.dualDirDpadMode) {
    uint8_t gamepadDpad = gpadToBinary(gamepad->options.dpadMode, gamepad->state);
    if ( socdMode == SOCD_MODE_NEUTRAL ) {
        dualOut = SOCDCombine(socdMode, gamepadDpad);
    } else if ( socdMode != SOCD_MODE_BYPASS ) {
        dualOut = SOCDGamepadClean(dualOut | gamepadDpad, socdMode == SOCD_MODE_SECOND_INPUT_PRIORITY);
    } else {
        dualOut |= gamepadDpad;
    }
}```

I think line 158 to 161 will be replaced by this change.

**Comment by deeebug** [comment]
Thank you for the update, looking good to me!
I've done some tests and suggested a bit of a change, so please look into it.

**Comment by bsstephan** [comment]
@deeebug I believe this also makes sense to me!

**Review by arntsonl — APPROVED** [review]
Nice, SOCD with dual is so confusing but glad we're getting some solid fixes!

**Pull request merged**